src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int MULTIPLIER_CAPACITY_AND_THRESHOLD = 2;
+    private int threshold;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] nodes;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        threshold = (int) (capacity * LOAD_FACTOR);
+        nodes = new Node[capacity];
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
```

Comment:
```suggestion
        if (size == threshold) {
```
size will never be > than threshold. and we can put this check inside `increaseArray()` method.

Reply:
I redid it.